mne/io/eeglab/eeglab.py

```diff
@@ -253,15 +253,41 @@ def _set_dig_montage_in_init(self, montage):
         self.set_montage(montage + make_dig_montage(ch_pos=ch_pos, coord_frame="head"))
 
 
-def _handle_montage_units(montage_units):
-    n_char_unit = len(montage_units)
-    if montage_units[-1:] != "m" or n_char_unit > 2:
-        raise ValueError(
-            '``montage_units`` has to be in prefix + "m" format'
-            f', got "{montage_units}"'
-        )
+def _handle_montage_units(montage_units, eeg):
+    if montage_units == "auto":
+        # estimate units from the channel position value range
+        # units could be mm, cm, or m
+        if (
+            hasattr(eeg, "chanlocs")
+            and "X" in eeg.chanlocs
+            and "Y" in eeg.chanlocs
+            and "Z" in eeg.chanlocs
+            and len(eeg.chanlocs["X"]) > 0
+        ):
+            xyz = np.array([eeg.chanlocs["X"], eeg.chanlocs["Y"], eeg.chanlocs["Z"]]).T
+            is_nan_locs = np.isnan(xyz).any(axis=1)
+            mean_radius = np.mean(np.linalg.norm(xyz[~is_nan_locs], axis=1))
+            # radius should be between 0.05 and 0.11 meters
+            if mean_radius < 1:  # m
+                prefix = ""
+            elif mean_radius > 20:  # mm
+                prefix = "m"
+            else:  # cm
+                prefix = "c"
+        else:
+            raise ValueError(
+                "Could not determine the units of the channel positions "
+                "automatically. Please specify the ``montage_units`` parameter."
+            )
```

Comment:
This is a backward incompatible change -- previously for example if `len(eeg.chanlocs["X"]) == 0` it would default to `mm` and "just work". So that's probably the path that should be preserved here
```suggestion
        else:
            prefix = "mm"  # backward compatible assumption
```

Reply:
Fixed.